### PR TITLE
Add Config::add_root_certificate for trusting custom ca

### PR DESCRIPTION
**Pull request by iamjpotts**
Support connecting to registries with self-signed certificates and certificates signed by CAs that are not in the client system trust store.

Simply passes thru the certificate along to a `reqwest` builder method by the same name.

Adds two tests:

1. Verify the client fails with an SSL error when the CA is not provided
2. Verify the client connects without an SSL error when the CA is given

Both tests are a part of the `test-mock` feature.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ Cargo.lock
 .envrc
 default.nix
 shell.nix
+.idea
```
Comment (PratikMahajan) [inline .gitignore:6]: this need not be in the commit. Depends on dev workflow, ide.
Can we worked around by using gobal .gitignore on dev system

**Review by PratikMahajan — CHANGES_REQUESTED** [review]
PR needs rebase

File: .gitignore (hunk 1 above)
Comment (iamjpotts) [reply .gitignore]: Removed.

**Comment by iamjpotts** [comment]
> PR needs rebase

Surprised, what did you notice that made it look like it needed a rebase? It was rebased about 5 months ago.

It should be current with the default branch now in any case.

**Comment by iamjpotts** [comment]
When this PR was originally authored 2 years ago, `hyper 1.x` had not been released.

Do you want me to upgrade this PR to hyper 1.x, or leave it on version 0.14?

**Comment by iamjpotts** [comment]
> can you upgrade to hyper 1.x and it'll be good to go!

Done.

**Review by PratikMahajan — APPROVED** [review]
/lgtm

**Comment by PratikMahajan** [comment]
looks like we need to move to [v0.14.28](https://github.com/hyperium/hyper/releases/tag/v0.14.28)
sorry about that

**Comment by iamjpotts** [comment]
* Back to hyper 0.14
* Fixed `cargo fmt` issues

**Pull request merged**